File: sprint1/Java/lesson7code4.java
public static boolean[] eratosthenes(int n) {
    boolean[] numbers = new boolean[n + 1];
    Arrays.fill(numbers, true);
    numbers[0] = false;
    numbers[1] = false;
    for (int num = 2; num < n; num++) {
        if (numbers[num]) {
            for (int j = 2 * num; j < n + 1; j += num) {
                numbers[j] = false;
            }
        }
    }
    return numbers;
}

File: sprint1/C#/lesson7code4.cs
public static bool[] Eratosthenes(int n)
{
    bool[] numbers = Enumerable.Range(0, n + 1).Select(i => true).ToArray();
    numbers[0] = numbers[1] = false;

    for (int num = 2; num < n; num++)
    {
        if (numbers[num])
        {
            for (int j = 2 * num; j <= n; j += num)
            {
                numbers[j] = false;
            }
        }
    }

    return numbers;
}
